src/Components/Components/src/RouteView.cs

```diff
@@ -82,35 +70,49 @@ protected virtual void Render(RenderTreeBuilder builder)
 
         builder.OpenComponent<LayoutView>(0);
         builder.AddComponentParameter(1, nameof(LayoutView.Layout), pageLayoutType);
-        builder.AddComponentParameter(2, nameof(LayoutView.ChildContent), _renderPageWithParametersDelegate);
+        builder.AddComponentParameter(2, nameof(LayoutView.ChildContent), RenderPageWithParameters);
         builder.CloseComponent();
     }
 
     private void RenderPageWithParameters(RenderTreeBuilder builder)
     {
-        builder.OpenComponent(0, RouteData.PageType);
+        var pathStart = NavigationManager.BaseUri.Length;
+        var name = NavigationManager.Uri.Substring(
+            pathStart,
+            NavigationManager.Uri.AsSpan().IndexOfAny("?#") switch
+            {
+                -1 => NavigationManager.Uri.Length - pathStart,
+                var index => index - pathStart
+            });
```

Comment:
Forms bind inside a "context", which has a name. "RouteView" establishes a "default context" matching the path.